graphql-ballerina/Module.md

```diff
@@ -0,0 +1,685 @@
+## Overview
+
+This module provides an implementation for connecting and interacting with GraphQL endpoints.
+
+GraphQL is an open-source data query and manipulation language for APIs. GraphQL allows clients to define the structure of the data required, and the same structure of the data is returned from the server, therefore preventing the returning of excessively large amounts of data.
+
+The Ballerina GraphQL implementation is using HTTP as the underlying protocol.
+
+### Listener
+
+The `graphql:Listener` is used to listen to a given IP/Port. To create a `graphql:Listener`, an `http:Listener` or a port number can be used.
+
+#### Create a standalone `graphql:Listener`
+```ballerina
+import ballerina/graphql;
+
+listener graphql:Listener graphqlListener = new(4000);
+```
+
+#### Create a `graphql:Listener` using an `http:Listener`
+```ballerina
+import ballerina/graphql;
+import ballerina/http;
+
+listener http:Listener httpListener = check new(4000);
+listener graphql:Listener graphqlListener = new(httpListener);
+```
+
+#### Additional Configurations
+When initializing the Ballerina GraphQL listener, a set of additional configurations can be provided to configure the listener including security and resiliency settings.
+The configurations that can be passed here are defined in the `graphql:ListenerConfiguration` record.
+
+```ballerina
+import ballrina/graphql;
+
+listener graphql:Listener graphqlListener = new (4000, timeout = 10, secureSocket = { key: { path: <KEYSTORE_PATH>, password: <PASSWORD>}});
+```
+
+### Service
+The Ballerina GraphQL service represents the GraphQL schema. When a service is attached to a `graphql:Listener`, a GraphQL schema will be auto-generated. The resource functions inside the service represent the resolvers of the root type.
+Then, the GraphQL listener will handle all the incoming requests and dispatch them to the relevant resource function.
+
+Since the GraphQL endpoints are exposed through a single endpoint, the endpoint URL of the GraphQL service can be provided after the service declaration, as shown in the following code snippet, where the endpoint URL is `/graphql`.
```

Comment:
```suggestion
Since the GraphQL endpoints are exposed through a single endpoint, the endpoint URL of the GraphQL service can be provided after the service declaration as shown in the following code snippet in which the endpoint URL is `/graphql`.
```